test/functional/p2p_leak.py

```diff
@@ -106,17 +94,24 @@ def set_test_params(self):
         self.num_nodes = 1
 
     def run_test(self):
+        # Peer that never sends a version. We will send a bunch of messages
+        # from this node anyway and verify eventual disconnection.
         no_version_disconnect_node = self.nodes[0].add_p2p_connection(
-            CNodeNoVersionMisbehavior(), send_version=False, wait_for_verack=False)
-        no_version_idlenode = self.nodes[0].add_p2p_connection(CNodeNoVersionIdle(), send_version=False, wait_for_verack=False)
+            CLazyNode(), send_version=False, wait_for_verack=False)
+
+        # Another peer that never sends a version. Just sits idle and hopes to receive
```

Comment:
I like the documentation you've added!
nit: I think "hopes" kind of sounds like it's doing something actively
```suggestion
        # Another peer that never sends a version, nor any other messages. It shouldn't receive anything from the node.
```